lib/src/extensions.dart

```diff
@@ -1,19 +1,33 @@
+enum RTCIceTransportPolicy {
+  all,
+  relay,
+}
+
+extension RTCIceTransportPolicyExt on RTCIceTransportPolicy {
+  String toStringValue() => {
+        RTCIceTransportPolicy.all: 'all',
+        RTCIceTransportPolicy.relay: 'relay',
+      }[this]!;
+}
+
 class RTCConfiguration {
   int? iceCandidatePoolSize;
   List<RTCIceServer>? iceServers;
-  String? iceTransportPolicy;
+  RTCIceTransportPolicy? iceTransportPolicy;
 
   Map<String, dynamic> toMap() {
-    final iceServersMap = <Map<String, dynamic>>[];
-    for (final element in (iceServers ?? <RTCIceServer>[])) {
-      iceServersMap.add(element.toMap());
-    }
+    //
+    final iceServersMap = <Map<String, dynamic>>[
+      if (iceServers != null)
+        for (final element in iceServers!) element.toMap()
+    ];
+
     return <String, dynamic>{
       // only supports unified plan
       'sdpSemantics': 'unified-plan',
+      if (iceServersMap.isNotEmpty) 'iceServers': iceServersMap,
       if (iceCandidatePoolSize != null) 'iceCandidatePoolSize': iceCandidatePoolSize,
-      'iceServers': iceServersMap,
-      if (iceTransportPolicy != null) 'iceTransportPolicy': iceTransportPolicy,
+      if (iceTransportPolicy != null) 'iceTransportPolicy': iceTransportPolicy!.toStringValue(),
```

Comment:
just an observation here. interesting the `!` is still needed even when it's not null

Reply:
Yah... not so cool right? 😅